### PR TITLE
Initial enemy setup - Slasher

### PR DESCRIPTION
**Pull request by joshualeecoss**
Basic scripts and prefab for the Slasher enemy, including Move and Idle state transitions, and ScriptableObject for enemy stats

### CODE REVIEW

**Comment by joshualeecoss** [comment]
[MS2-1: Initial Enemy Setup](https://trello.com/c/7vj8iNgS/12-ms2-1-initial-enemy-setup)

**Comment by joshualeecoss** [comment]
[MS2-1: Initial Enemy Setup (Slasher)](https://trello.com/c/StR4nJCX/54-ms2-1-initial-enemy-setup-slasher)

File: Assets/Scripts/Player/PlayerController.cs [hunk 1]
```diff
@@ -10,51 +10,59 @@
...
+    private PlayerStateMachine playerStateMachine;
     public Rigidbody2D rb;
     // Reference to player stats script
     private PlayerStats player;
 
     private float moveSpeed;
+    private float attackSpeed;
+    private float strength;
+    private float healthPoints;
     // Vector2 of the normalized vectors of movement for the player
```
Comment (josephlatina) [inline Assets/Scripts/Player/PlayerController.cs:28]: Are we thinking of including the other player attributes at a later time? ie. Defence, Incoming Damage and the enum Player Status?

File: Assets/Scripts/Enemy/EnemyController.cs [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+    private float attackSpeed;
+
+    // Strength: amount of damage inflicted by the enemy
+    private float strength;
+
+    // Health points of the enemy
+    private float healthPoints;
+
+    // Status effect of the enemy (placeholder enum, could be expanded)
+    private enum EnemyStatus { };
```
Comment (josephlatina) [inline Assets/Scripts/Enemy/EnemyController.cs]: We could probably already define the enums for Enemy Status here: Stunned, Immobilized, Poisoned and Normal (refer to documentation)

**Review by josephlatina — COMMENTED** [review]
Overall looks good! Just some minor comments that need to be addressed :)

File: Assets/Scripts/Player/PlayerController.cs (hunk 1 above)
Comment (joshualeecoss) [reply Assets/Scripts/Player/PlayerController.cs:28]: That was my thinking, but I can add them if needed

Comment (joshualeecoss) [reply Assets/Scripts/Player/PlayerController.cs:28]: I added them

File: Assets/Scripts/Enemy/EnemyController.cs (hunk 2 above)
Comment (joshualeecoss) [reply Assets/Scripts/Enemy/EnemyController.cs]: Added

**Review by josephlatina — APPROVED** [review]
Looks good!

**Pull request merged**